docs/essentials/calendar.md

```diff
@@ -0,0 +1,147 @@
+ms.author: ScottYoung-5180
+ms.date: 03/12/2019
+---
+
+# Xamarin.Essentials: Calendar
+
+The **Calendar** class lets you view the user's calendar and event information.
+
+## Get started
+
+[!include[](~/essentials/includes/get-started.md)]
+
+To access the **Calendar** functionality the following platform specific setup is required.
+
+# [Android](#tab/android)
+
+The `Read Calendar` and `Write Calendar` permission is required and must be configured in the Android project. 
+
+Open the **AndroidManifest.xml** file under the **Properties** folder and add the following inside of the **manifest** node.
+
+```xml
+<uses-permission android:name="android.permission.READ_CALENDAR" />
+```
+
+Or right click on the Android project and open the project's properties. Under **Android Manifest** find the **Required permissions:** area and check the **READ_CALENDAR** permission. This will automatically update the **AndroidManifest.xml** file.
+
+# [iOS](#tab/ios)
+
+Your app's **Info.plist** must contain the `NSCalendarsUsageDescription` key in order to access the device’s calendars & reminders.
+
+Open the plist editor and add the **Privacy - Calendars Usage Description** property and fill in a value to display to the user.
+
+Or manually edit the file and add the following and update the rationale:
+
+```xml
+<key>NSCalendarsUsageDescription</key>
+<string>Fill in a reason why your app needs access to calendars.</string>
+```
+
+# [UWP](#tab/uwp)
+
+You must set the `Appointments` permission for the application. This can be done by opening the **Package.appxmanifest** and selecting the **Capabilities** tab and checking **Appointments**.
+
+-----
+
+## Using Calendar
+
+Add a reference to Xamarin.Essentials in your class:
+
+```csharp
+using Xamarin.Essentials;
+```
+
+```
+> [!NOTE]
+> When no dates are set for event retrieval the default will retrieve from now --> 14 days from now
+```
+
+
+Get all existing calendars:
+
+```csharp
+var calendars = await Calendar.GetCalendarsAsync(); 
+// The var calendars will now hold a list of all existing/retrievable calendars. (List<DeviceCalendar>)
+```
+
+Get all existing events for all calendars using default date range:
+```
+> [!TIP]
+> When pulling back events they use an 'overlapping' formula, i.e. if I had an event that went from 9:00 - 10:00 and my start or end time intersects that time slot it will be included in the resulting list. 
+```
+```csharp
+var events =  await Calendar.GetEventsAsync(); 
+// The var events will now hold a list of existing/retrievable events for all calendars. (List<DeviceEvent>)
+```
+
+Get all existing events for a specific calendar using default date range:
+
+```csharp
+using System.Linq;
+
+try
+{
+	var calendarId = (await Calendar.GetCalendarsAsync()).First().Id;
+	var events =  await Calendar.GetEventsAsync(calendarId); 
+	// The var events will now hold a list of existing/retrievable events for our first calendar. (List<DeviceEvent>)
+}
+catch (ArgumentNullException)
```

Comment:
Why this type of exception? This should only be thrown if the argument (eg calendarId) is null

Reply:
Ahh I think my comment is misleading, if you were get get 0 events back no exception would be thrown, the only way it gets an exception is if you supply an id that doesn't exist, and I realise the code has changed so it should be expecting a ArgumentException instead of ArgumentNullException

Reply:
Ok, we should clarify this. Does our code always through an ArgumentException (ie for all platforms)?

Reply:
See changes in new PR:
https://github.com/builttoroam/xamarin-docs/pull/2/files